src/reconcile/rec_visibility.c

```diff
@@ -564,27 +234,27 @@ __wt_rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins, v
              * to discard updates from the stable update and older for correctness and we can't
              * discard an uncommitted update.
              */
-            if (F_ISSET(r, WT_REC_UPDATE_RESTORE) && walked_past_sel_upd &&
+            if (F_ISSET(r, WT_REC_UPDATE_RESTORE) && upd_select->upd != NULL &&
               (list_prepared || list_uncommitted))
-                WT_ERR(__wt_set_return(session, EBUSY));
-
-            if (upd->type == WT_UPDATE_BIRTHMARK)
-                skipped_birthmark = true;
+                return (__wt_set_return(session, EBUSY));
 
             /*
              * Track the oldest update not on the page.
              *
              * This is used to decide whether reads can use the page image, hence using the start
              * rather than the durable timestamp.
              */
-            if (!walked_past_sel_upd && upd->start_ts < r->min_skipped_ts)
+            if (upd_select->upd == NULL && upd->start_ts < r->min_skipped_ts)
                 r->min_skipped_ts = upd->start_ts;
+        } else if (first_stable_upd == NULL) {
```

Comment:
when the newest version is written min_skipped_ts is always supposed to be WT_TS_MAX.